app/src/main/java/net/bible/android/view/activity/search/MySearchResults.kt

```diff
@@ -0,0 +1,438 @@
+                    verseTextSpannable.getChars(start, end, wordArray, 0)
+                    val originalWord = wordArray.joinToString("")
+                    val word = originalWord.lowercase()
+                    val wordStat = wordStatistics.firstOrNull{it.word == word}
+                    if (wordStat == null) {
+                        wordStatistics.add(WordStat(word, intArrayOf(listIndex), originalWord))
+                    } else {
+                        wordStatistics.first{it.word == word}.verseIndexes += listIndex
+                    }
+                }
+            }
+            wordStatsTimer.stop()
+            listIndex += 1
+        }
+        sectionsPagerAdapter.verseListFrag.arrayAdapter.notifyDataSetChanged()
+        sectionsPagerAdapter.getItem(bookTabPosition)
+        sectionsPagerAdapter.getItem(wordTabPosition)
+        sectionsPagerAdapter.notifyDataSetChanged()
+
+        val tabHost = this.findViewById<View>(R.id.tabs) as TabLayout
+        tabHost.getTabAt(verseTabPosition)!!.text = CommonUtils.resources.getString(R.string.verse_count, mSearchResultsArray.count().toString())  // For some reason the value set in 'setResultsAdapter' get's cleared so I need to do it here as well.
+        tabHost.getTabAt(bookTabPosition)!!.text = CommonUtils.resources.getString(R.string.book_count, bookStatistics.count().toString())
+        tabHost.getTabAt(wordTabPosition)!!.text = CommonUtils.resources.getString(R.string.word_count, wordStatistics.count().toString())
+
+        populateViewResultsAdapterTimer.stop()
+        populateViewResultsAdapterTimer.log()
+        getSearchResultVerseElementTimer.log()
+        getSearchResultVerseStringTimer.log()
+        verseTextSpannableTimer.log()
+        bookStatsTimer.log()
+        wordStatsTimer.log()
+        totalTime.stop(true)
+    }
+    /**
+     * Handle scripture/Appendix toggle
+     */
+    private val scriptureToggleClickListener = View.OnClickListener {
+        isScriptureResultsCurrentlyShown = !isScriptureResultsCurrentlyShown
+        populateViewResultsAdapter()
+//        mKeyArrayAdapter!!.notifyDataSetChanged()
+        searchResultsActionBarManager.setScriptureShown(isScriptureResultsCurrentlyShown)
+    }
+
+    companion object {
+        private const val TAG = " MySearchResults"
+        private const val LIST_ITEM_TYPE = android.R.layout.simple_list_item_2
+    }
+}
+
+class SearchResultsPagerAdapter(private val context: Context, fm: FragmentManager,
+                                searchControl: SearchControl,
+                                activeWindowPageManagerProvider: ActiveWindowPageManagerProvider,
+                                intent: Intent,
+                                val mSearchResultsArray:ArrayList<SearchResultsData>,
+                                val bookStatistics: MutableList<BookStat>,
+                                val wordStatistics: MutableList<WordStat>
```

Comment:
we could pass activity as context and avoid long list of parameters

Reply:
This was my first attempt at pages and fragments and other stuff. So there is a lot in here that needs fixing or improving. I think it might almost be better for someone who knows what they are doing just to start again.